server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -6,81 +6,71 @@ import uk.gov.ons.addressIndex.server.model.dao.ElasticClientProvider
 @ImplementedBy(classOf[AddressIndexRepository])
-trait ElasticsearchRepository extends ElasticIndexSugar {
+trait ElasticsearchRepository {
   /**
     * An ElasticClient.
     */
   def client() : ElasticClient
 
   /**
-    * Create the repository.
+    * Query the PAF addres index by UPRN.
     *
-    * @return
+    * @param uprn the identificator of the address
+    * @return Fucture containing a PAF address or `None` if not in the index
     */
-  def createAll() : Future[Seq[_]]
+  def queryPafUprn(uprn : String) : Future[Option[PostcodeAddressFileAddress]]
 
   /**
-    * Delete the repository.
+    * Query the NAG addres index by UPRN.
     *
-    * @return
+    * @param uprn the identificator of the address
+    * @return Fucture containing a NAG address or `None` if not in the index
     */
-  def deleteAll() : Future[Seq[_]]
+  def queryNagUprn(uprn : String) : Future[Option[NationalAddressGazetteerAddress]]
 
   /**
-    * Query the addres index by UPRN.
+    * Query the address index for PAF addresses.
+    * Currently the query must be for building number and postcode.
     *
-    * @param uprn
-    * @return
+    * @param tokens address tokens
+    * @return Future with found PAF addresses and the maximum score
     */
-  def queryUprn(uprn : String) : Future[Option[PostcodeAddressFileAddress]]
+  def queryPafAddresses(tokens: AddressTokens) : Future[PostcodeAddressFileAddresses]
 
   /**
-    * Query the address index for addresses.
+    * Query the address index for NAG addresses.
     * Currently the query must be for building number and postcode.
     *
     * @param tokens address tokens
-    * @return Future with found addresses and the maximum score
+    * @return Future with found PAF addresses and the maximum score
     */
-  def queryAddress(tokens: AddressTokens) : Future[PostcodeAddressFileAddresses]
+  def queryNagAddresses(tokens: AddressTokens) : Future[NationalAddressGazetteerAddresses]
 }
 
 @Singleton
 class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticClientProvider: ElasticClientProvider)(implicit ec: ExecutionContext) extends ElasticsearchRepository {
 
-  private val logger = Logger("address-index:ElasticsearchRepositoryModule")
   private val esConf = conf.config.elasticSearch
   private val pafIndex = esConf.indexes.pafIndex
+  private val nagIndex = esConf.indexes.nagIndex
   val client: ElasticClient = elasticClientProvider.client
 
-  def createAll() : Future[Seq[CreateIndexResponse]] = {
-    createIndex(
-      PostcodeAddressFileAddress,
-      PostcodeIndex
-    )(client)
-  }
-
-  def deleteAll() : Future[Seq[DeleteIndexResponse]] = {
-    deleteIndex(
-      PostcodeAddressFileAddress,
-      PostcodeIndex
-    )(client)
-  }
-
-  def queryUprn(uprn: String): Future[Option[PostcodeAddressFileAddress]] = client.execute{
+  def queryPafUprn(uprn: String): Future[Option[PostcodeAddressFileAddress]] = client.execute{
     search in pafIndex query { termQuery("uprn", uprn) }
```

Comment:
As saw together, this is not as trivial to fix as we thought. It's not cost effective to spend too much time on it and the fix may add more lines than it removes, so IMO this is good enough, I won't modify it any more.